### DISCUSSION TITLE
Support for on-the-fly PDF generation

### DISCUSSION THREAD

**Discussion by renaudpawlak** [cincheo/daquota#48] [Announcements]
From version 0.9.23, each component provides a ``downloadAsPDF`` action, which generates and downloads the PDF corresponding to that component.

It uses the html2pdf.js client-side rendering lib. See the supported options here: https://github.com/eKoopmans/html2pdf.js#options

See the invoice app for an example of use (generating PDF files for invoices).

**Comment by HFolkertH** [comment]
Hi Renaud,

i have a new job so less time to experiment with dliteful.

for reporting www.apitemplate.io is also very nice. 50 runs for free.
should be possible to connect dlite to the rest api of apitemplate.io?


regards
Folkert

**Reply by renaudpawlak** [reply]
Well, I hope your new job is going well :) Still planing to finish the UI benchmark experiment?
I have picked html2pdf.js because it is free, Open Source (MIT licence), and it runs in the browser without any external API. Plus, it is very simple to use because it converts an element of the web page to a PDF. That said, any one can use another API if it is REST accessible. Just use an HTTP endpoint to invoke it. We could also create a connector to be installed as a plugin for that sort of service if it is worth doing it.
